wsd/Admin.cpp

```diff
@@ -685,6 +691,28 @@ void Admin::pollingThread()
         LOG_TRC("Admin poll for " << timeout);
         poll(timeout); // continue with ms for admin, settings etc.
     }
+
+    if (!COOLWSD::IndirectionServerEnabled)
+        return;
+
+    _model.sendShutdownReceivedMsg();
+
+    constexpr size_t sleepMs = 1000;
+    const auto migrationMsgTimeout =
+        COOLWSD::getConfigValue<int>("indirection_endpoint.migration_timeout_secs", 180) * 1000;
+    size_t count = migrationMsgTimeout / sleepMs;
+    for (size_t i = 0; i < count; ++i)
+    {
+        poll(std::chrono::seconds(0));
```

Comment:
Why not pass `sleepMs` to `poll`? If it has no events, it will sleep for the full duration. If some even comes through, it'll wake. Of course you would also need to replace the fixed-iterations loop with a duration loop (i.e. you would need to track how long it has passed since you started the loop, and break it when you exceed `migrationMsgTimeout`).